File: LeetCode/cpp/806.cpp
class Solution {
public:
    vector<int> numberOfLines(vector<int> &widths, string &S) {
        int line = 1, width = 0;
        for (auto s:S) {
            if (width + widths[s - 'a'] <= 100)
                width += widths[s - 'a'];
            else {
                width = widths[s - 'a'];
                ++line;
            }
        }
        return vector<int>{line, width};
    }
};

File: LeetCode/python3/806.py
class Solution:
    def numberOfLines(self, widths: List[int], S: str) -> List[int]:
        line, last_width = 1, 0
        for s in S:
            width = widths[ord(s) - ord('a')]
            if last_width + width <= 100:
                last_width += width
            else:
                last_width = width
                line += 1
        return [line, last_width]
